pkg/controller/direct/privilegedaccessmanager/entitlement_controller.go

```diff
@@ -283,15 +264,20 @@ func (a *Adapter) Create(ctx context.Context, createOp *directbase.CreateOperati
 	mapCtx := &direct.MapContext{}
 
 	desired := a.desired.DeepCopy()
-	resource := PrivilegedAccessManagerEntitlementSpec_ToProto(mapCtx, &desired.Spec)
+	resourceType, resource, err := getResourceTypeAndResourceFromContainer(a.id.Parent.Container)
+	if err != nil {
+		return fmt.Errorf("error getting resourceType and resource from container: %w", err)
+	}
+	hiddenFields := gcpIAMAccessResource{resourceType: resourceType, resource: resource}
+	entitlement := PrivilegedAccessManagerEntitlementSpec_ToProto(mapCtx, &desired.Spec, hiddenFields)
```

Comment:
It looks weird if we pass in the hiddenFields to PrivilegedAccessManagerEntitlementSpec_ToProto() so I'm also exploring the option to improve the readability in #2893.